### PR TITLE
Expose structural GraphQL types

### PR DESCRIPTION
**Pull request by calebmer**
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12078, but on the `master` branch.

/cc @TonyPythoneer, @zhengbli

### CODE REVIEW

**Comment by TonyPythoneer** [comment]
@calebmer Would you want to maintain this type definition together? I want to invite you. :smile: 
I had looked your repo about [postgraphql/typings/graphql.d.ts](https://github.com/calebmer/postgraphql/blob/master/typings/graphql.d.ts).
I think you have experience about type assert for this.

If you have the will, would you open another PR about adding your name in authors? thx. :smile:

**Comment by calebmer** [comment]
I’d love to, see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12179 :blush:

**Comment by TonyPythoneer** [comment]
@calebmer Would you take a look these reviews if you have time? thx.

**Comment by calebmer** [comment]
I’m not sure I understand?

**Comment by TonyPythoneer** [comment]
@calebmer If you have any question, please answer my reviews. I will explain it.

**Comment by calebmer** [comment]
I don’t see an reviews @TonyPythoneer. Did you submit them?

**Comment by TonyPythoneer** [comment]
@calebmer sorry,it's my mistake. I hadn't clicked to submit. Now, have you seen?

File: graphql/graphql.d.ts [hunk 1]
```diff
@@ -1552,36 +1468,36 @@ declare module "graphql/type/definition" {
...
+    export interface GraphQLInputObjectField {
         name: string;
         type: GraphQLInputType;
         defaultValue?: any;
         description?: string;
     }
 
-    interface InputObjectFieldMap {
-        [fieldName: string]: InputObjectField;
+    export interface GraphQLInputObjectFieldMap {
```
Comment (calebmer) [inline graphql/graphql.d.ts]: I’d argue that in the interest of keeping type definition quality high, we don’t need to match the source code exactly. We should definitely need to exactly match functionality, but in the names of type aliases we have a bit more freedom to improve upon the source code itself.

I’d also argue that the intention for these types is to be named with a ‘GraphQL’ prefix given the other types and the names probably were meant to align with the `GraphQLObjectType` types.

I’ve opened a PR here to change the source code https://github.com/graphql/graphql-js/pull/526, we’ll see how that goes. In the meantime I can change these back if you’d like :+1:

Comment (TonyPythoneer) [inline graphql/graphql.d.ts]: I agree your comment on graphql-js.

File: graphql/graphql.d.ts [hunk 2]
```diff
@@ -19,69 +19,7 @@ declare module "graphql" {
...
-        isInputType,
-        isOutputType,
-        isLeafType,
-        isCompositeType,
-        isAbstractType,
-
-        // Un-modifiers
-        getNullableType,
-        getNamedType,
-    } from 'graphql/type';
```
Comment (TonyPythoneer) [inline graphql/graphql.d.ts]: @calebmer Would comment this? I want to understand your thinking.

File: graphql/graphql.d.ts (hunk 1 above)
Comment (TonyPythoneer) [inline graphql/graphql.d.ts]: @calebmer Great! I had seen your [PR](https://github.com/graphql/graphql-js/pull/526) get merged.

File: graphql/graphql.d.ts (hunk 2 above)
Comment (TonyPythoneer) [inline graphql/graphql.d.ts]: @calebemr About thinking about this, I shouldn't focus formal of exports .

**Review by TonyPythoneer — APPROVED** [review]

**Comment by TonyPythoneer** [comment]
@vvakame Would you merge this? thx.

**Comment by calebmer** [comment]
The names I changed in https://github.com/graphql/graphql-js/pull/526 aren’t the same as the ones I initially changed here. They’ve been updated to match the source code. It should be good to merge :+1:

**Comment by TonyPythoneer** [comment]
Great! This PR had got resolved.

**Pull request merged**